cmd/util/cmd/epochs/cmd/recover.go

```diff
@@ -0,0 +1,186 @@
+// EFM can be exited only by a special service event, EpochRecover, which initially originates from a manual service account transaction.
+// The full epoch data must be generated manually and submitted with this transaction in order for an
+// EpochRecover event to be emitted. This command retrieves the current protocol state identities, computes the cluster assignment using those
+// identities, generates the cluster QC's and retrieves the DKG key vector of the last successful epoch.
+var (
+	generateRecoverEpochTxArgsCmd = &cobra.Command{
+		Use:   "generate-efm-recovery-data",
+		Short: "Generates recover epoch transaction arguments",
+		Long:  "Generates transaction arguments for the epoch recovery transaction.",
+		Run:   generateRecoverEpochTxArgs,
+	}
+
+	flagAnAddress               string
+	flagAnPubkey                string
+	flagPartnerWeights          string
+	flagPartnerNodeInfoDir      string
+	flagInternalNodePrivInfoDir string
+	flagConfig                  string
+	flagCollectionClusters      int
+	flagStartView               uint64
+	flagStakingEndView          uint64
+	flagEndView                 uint64
+)
+
+func init() {
+	rootCmd.AddCommand(generateRecoverEpochTxArgsCmd)
+	addGenerateRecoverEpochTxArgsCmdFlags()
+}
+
+func addGenerateRecoverEpochTxArgsCmdFlags() {
+	generateRecoverEpochTxArgsCmd.Flags().StringVar(&flagBucketNetworkName, "bucket-network-name", "",
+		"when retrieving the root snapshot from a GCP bucket, the network name portion of the URL (eg. \"mainnet-13\")")
+	generateRecoverEpochTxArgsCmd.Flags().IntVar(&flagCollectionClusters, "collection-clusters", 0,
+		"number of collection clusters")
+	// required parameters for network configuration and generation of root node identities
+	generateRecoverEpochTxArgsCmd.Flags().StringVar(&flagConfig, "config", "",
+		"path to a JSON file containing multiple node configurations (fields Role, Address, Weight)")
+	generateRecoverEpochTxArgsCmd.Flags().StringVar(&flagInternalNodePrivInfoDir, "internal-priv-dir", "", "path to directory "+
+		"containing the output from the `keygen` command for internal nodes")
+	generateRecoverEpochTxArgsCmd.Flags().StringVar(&flagPartnerNodeInfoDir, "partner-dir", "", "path to directory "+
+		"containing one JSON file starting with node-info.pub.<NODE_ID>.json for every partner node (fields "+
+		" in the JSON file: Role, Address, NodeID, NetworkPubKey, StakingPubKey)")
+	generateRecoverEpochTxArgsCmd.Flags().StringVar(&flagPartnerWeights, "partner-weights", "", "path to a JSON file containing "+
+		"a map from partner node's NodeID to their stake")
+	generateRecoverEpochTxArgsCmd.Flags().Uint64Var(&flagStartView, "start-view", 0, "start view of the recovery epoch")
+	generateRecoverEpochTxArgsCmd.Flags().Uint64Var(&flagStakingEndView, "staking-end-view", 0, "end view of the staking phase of the recovery epoch")
+	generateRecoverEpochTxArgsCmd.Flags().Uint64Var(&flagEndView, "end-view", 0, "end view of the recovery epoch")
```

Comment:
I think it would be easier from an ops perspective if we used the same representation for these values as we do in bootstrapping, in particular using the lengths as the input rather than the end view. I suggest we use the same flag names as the [`rootblock` comand](https://github.com/onflow/flow-go/blob/master/cmd/bootstrap/cmd/rootblock.go#L77-L79):
- `--epoch-length`
- `--epoch-staking-phase-length`

Then we can compute `endView` and `stakingEndView` as we do in the bootstrapping.

Reply:
https://github.com/onflow/flow-go/pull/5576/commits/884fdbf2dd8a1508f897819c55ea4e953287efa2